index.js

```diff
@@ -539,19 +539,33 @@ Router.prototype.find = function find (method, path, derivedConstraints) {
   }
 }
 
-Router.prototype._getWildcardNode = function (node, path, len) {
+Router.prototype._getWildcardNode = function (node, path, len, derivedConstraints, params) {
   if (node === null) return null
   var decoded = fastDecode(path.slice(-len))
   if (decoded === null) {
     return this.onBadUrl !== null
       ? this._onBadUrl(path.slice(-len))
       : null
   }
-  var handle = node.handlers[0]
+
+  var handle = derivedConstraints !== undefined ? node.getMatchingHandler(derivedConstraints) : node.unconstrainedHandler
```

Comment:
I think we should be using the same pattern like https://github.com/delvedor/find-my-way/blob/master/index.js#L380 for consistence behavior. And only `unconstrainedHandler` contains what we needs to compute `paramsObj`.